CalculatorLibrary/CalculatorLibrary.cs

```diff
@@ -0,0 +1,67 @@
+﻿// CalculatorLibrary.cs
+namespace CalculatorLibrary;
+
+using Newtonsoft.Json;
+public class Calculator
+{
+    JsonWriter _writer;
+    public Calculator()
+    {
+        StreamWriter logFile = File.CreateText("Calculator.json");
+       
+        _writer = new JsonTextWriter(logFile);
+        _writer.Formatting = Formatting.Indented;
+        _writer.WriteStartObject();
+        _writer.WritePropertyName("Operations");
+        _writer.WriteStartArray();
+        
+    }
+    public  double DoOperation(double num1, double num2, string op)
+    {
+        double result = double.NaN; // Default value is "not-a-number" if an operation, such as division, could result in an error.
+        _writer.WriteStartObject();
+        _writer.WritePropertyName("Operand1");
+        _writer.WriteValue(num1);
+        _writer.WritePropertyName("Operand2");
+        _writer.WriteValue(num2);
+        _writer.WritePropertyName("Operation");
+        // Use a switch statement to do the math.
+        switch (op)
+        {
+            case "a":
+                result = num1 + num2;
+                _writer.WriteValue("Add");
+                break;
+            case "s":
+                result = num1 - num2;
+                _writer.WriteValue("Subtract");
+                break;
+            case "m":
+                result = num1 * num2;
+                _writer.WriteValue("Multiply");
+                break;
+            case "d":
+                // Ask the user to enter a non-zero divisor.
+                if (num2 != 0)
+                {
+                    result = num1 / num2;
+                    _writer.WriteValue("Divide");
```

Comment:
This line: `_writer.WriteValue("Divide");` needs to be outside this if statement.
Currently if you test and input (for example) `1`, `0` and `d` you get an exception instead of the desired result of `"This operation will result in a mathematical error.`:

![image](https://github.com/user-attachments/assets/48c1bb19-70d3-4ac6-9794-c018e6044809)

Reply:
Good morning sir. thanks for you effor reviewing my code and will apply all of these advices in the coming projects. I don't understand what exactly is the problem in the could you please clarify more